src/cleaning/alru.c

```diff
@@ -517,7 +517,7 @@ static int ocf_alru_recovery_handle(ocf_parallelize_t parallelize,
 	int i;
 
 	portion = DIV_ROUND_UP((uint64_t)entries, shards_cnt);
-	begin = portion*shard_id;
+	begin = (uint64_t)portion*shard_id;
 	end = OCF_MIN(portion*(shard_id + 1), entries);
```

Comment:
here such cast is actually required to avoid overflow in portion * (shard_id + 1)